tools/diff-processor/cmd/add_labels.go

```diff
@@ -0,0 +1,115 @@
+package cmd
+
+import (
+	newProvider "google/provider/new/google/provider"
+	oldProvider "google/provider/old/google/provider"
+
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/magic-modules/tools/diff-processor/diff"
+	"github.com/GoogleCloudPlatform/magic-modules/tools/diff-processor/labels"
+	"github.com/GoogleCloudPlatform/magic-modules/tools/issue-labeler/labeler"
+	"github.com/spf13/cobra"
+	"golang.org/x/exp/maps"
+)
+
+const addLabelsDesc = `Add labels to a PR based on changed resources.`
+
+type addLabelsOptions struct {
+	rootOptions       *rootOptions
+	computeSchemaDiff func() diff.SchemaDiff
+	enrolledTeamsYaml []byte
+	prId              uint64
+	getIssue          func(repository string, id uint64) (labeler.Issue, error)
+	updateIssues      func(repository string, issueUpdates []labeler.IssueUpdate, dryRun bool)
+	dryRun            bool
+}
+
+func newAddLabelsCmd(rootOptions *rootOptions) *cobra.Command {
+	o := &addLabelsOptions{
+		rootOptions: rootOptions,
+		computeSchemaDiff: func() diff.SchemaDiff {
+			return diff.ComputeSchemaDiff(oldProvider.ResourceMap(), newProvider.ResourceMap())
+		},
+		enrolledTeamsYaml: labeler.EnrolledTeamsYaml,
+		getIssue:          labels.GetIssue,
+		updateIssues:      labeler.UpdateIssues,
+	}
+	cmd := &cobra.Command{
+		Use:   "add-labels PR_ID [--dry-run]",
+		Short: addLabelsDesc,
+		Long:  addLabelsDesc,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return errors.New("Missing pull request ID.")
+			}
+
+			prId, err := strconv.ParseUint(args[0], 10, 0)
+
+			if err != nil {
+				return fmt.Errorf("PR_ID must be an unsigned integer: %w", err)
+			}
+
+			o.prId = prId
```

Comment:
I went back and forth on this - flag validation persists its values on the options object, so I think persisting args on the options object could also be fine, but I don't have a strong opinion about it. It also turns out that doing validation at the top of RunE and returning an error is functionally identical to returning an error from Args. I don't want to actually do the parsing in RunE because it's more difficult to test and makes it less clear what's happening, so I've moved the validation into o.run() in 69ca5ef7b03b1d09a1ec0d69191ca47f314e55d4.